src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -27,155 +20,193 @@ def _calc_min_max_from_sim_dict(subdict: Dict, key: str):
+        create_or_update_subdict(
+            subdict, key_name,
+            {time_slot: core_stats[area["uuid"]][time_slot][key_name]}
+        )
 
         cls._calc_min_max_from_sim_dict(subdict, key_name)
 
-    def update(self, area):
+    def update(self, area, area_result_dict={}, core_stats={}):
         if self.should_export_plots:
-            self.gather_device_statistics(area, self.device_stats_dict, {})
+            self.gather_device_statistics(
+                area_result_dict, self.device_stats_dict, {}, core_stats)
         else:
-            self.gather_device_statistics(area, {}, self.current_stats_dict)
+            self.gather_device_statistics(
+                area_result_dict, {}, self.current_stats_dict, core_stats)
 
     @classmethod
-    def gather_device_statistics(cls, area: Area, subdict: Dict, flat_result_dict: Dict):
-        for child in area.children:
-            if child.name not in subdict.keys():
-                subdict.update({child.name: {}})
-            if child.children == []:
-                cls._gather_device_statistics(child, subdict[child.name], flat_result_dict)
+    def gather_device_statistics(cls, area: Dict, subdict: Dict,
+                                 flat_result_dict: Dict,
+                                 core_stats={}):
+        for child in area['children']:
+            if child['name'] not in subdict.keys():
+                subdict.update({child['name']: {}})
+            if child['children'] == []:
+                cls._gather_device_statistics(
+                    child, subdict[child['name']], flat_result_dict,
+                    core_stats)
             else:
-                cls.gather_device_statistics(child, subdict[child.name], flat_result_dict)
+                cls.gather_device_statistics(
+                    child, subdict[child['name']], flat_result_dict,
+                    core_stats)
 
     @classmethod
-    def _gather_device_statistics(cls, area: Area, subdict: Dict, flat_result_dict: Dict):
-        if not hasattr(area.parent, "past_markets") or len(area.parent.past_markets) == 0:
-            return None
-
-        if area.strategy is not None:
-            cls._device_price_stats(area, subdict)
-            cls._device_energy_stats(area, subdict)
-
-        if isinstance(area.strategy, PVStrategy):
-            cls._pv_production_stats(area, subdict)
-
-        elif isinstance(area.strategy, StorageStrategy):
-            cls._soc_stats(area, subdict)
-
-        elif isinstance(area.strategy, LoadHoursStrategy):
-            cls._load_profile_stats(area, subdict)
-
-        elif type(area.strategy) == FinitePowerPlant:
-            market = list(area.parent.past_markets)[-1]
-            create_or_update_subdict(subdict, "production_kWh",
-                                     {market.time_slot: area.strategy.energy_per_slot_kWh})
+    def _gather_device_statistics(cls, area: Dict, subdict: Dict,
```

Comment:
AFAIS, You pnly need the area_uuid here and in all sub_functions. So you can only parse this.

Reply:
Sorry, I didn't get it, I don't just need `area_uuid ` but also the area type here too, to distinguish between the type of data needs to be documented

Reply:
yes, we discussed this offline. You are right, but `area` is not the right variable name IMO. It should be `area_result_dict` (or similar) here and in all sub functions.

Reply:
`area_dict` would be sufficient actually.